src/doc/src/faq.md

```diff
@@ -259,3 +259,50 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### What does “version's conflict” mean and how to resolve it?
+
+> failed to select a version for `x` which could resolve this conflict
+
+Have you seen the error message above?
+
+This is one of the most annoying error message for Cargo users. There are several 
+situations may lead us to a version conflict. Below we'll walk through possible 
+causes and provide diagnostic techniques to help you out there:
+
+- The project and its dependencies use [links][links] to repeatedly link the local 
```

Comment:
Please follow the style adding two spaces for subsequent lines in the same list item.

```markdown
- First line.
  Second line.
- Another item.
```